### PR TITLE
[DG] ADY-38 As an Architect I would like to have a self registering Notification Receiver that queues notifications from Adyen Marketpay in the database so that they can be acknowledged quickly but processed asynchronously

### PR DESCRIPTION
**Pull request by pandemicGander**

### CODE REVIEW

File: src/main/java/com/adyen/mirakl/listeners/AdyenNotificationListener.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AdyenNotificationListener {
+
+    @Async
+    @EventListener
+    public void handleContextRefresh(AdyenNotifcationEvent event) {
+        System.out.println(event.getDbId());
```
Comment (lancergr) [inline src/main/java/com/adyen/mirakl/listeners/AdyenNotificationListener.java]: better log?

Comment (pandemicGander) [reply src/main/java/com/adyen/mirakl/listeners/AdyenNotificationListener.java]: Yeah, this was just a placeholder but I'll put a logger in there

File: src/main/java/com/adyen/mirakl/domain/AdyenNotification.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * A AdyenNotification.
+ */
+@Entity
+@Table(name = "adyen_notification")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
```
Comment (lancergr) [inline src/main/java/com/adyen/mirakl/domain/AdyenNotification.java]: is this put intentionally or added by default by jhipster?
it's not likely that we will get the same entity often - so I am not sure what the benefits of Cache annotation would be.. I believe a query cache would be fine

Comment (pandemicGander) [reply src/main/java/com/adyen/mirakl/domain/AdyenNotification.java]: Generated yeah, the cache might be useful on when and how we process the notifications as I publish an event with the primary key. 

But it's not a big deal either

**Pull request merged**